sklearn/utils/tests/test_validation.py

```diff
@@ -447,6 +447,25 @@ def test_check_array_pandas_na_support(pd_dtype, dtype, expected_dtype):
         check_array(X, force_all_finite=True)
 
 
+def test_check_array_panadas_na_support_series():
+    """Check check_array is correct with pd.NA in a series."""
+    pd = pytest.importorskip("pandas")
+
+    X_int64 = pd.Series([1, 2, pd.NA], dtype="Int64")
+
+    msg = "Input contains NaN"
+    with pytest.raises(ValueError, match=msg):
+        check_array(X_int64, force_all_finite=True, ensure_2d=False)
+
+    X_out = check_array(X_int64, force_all_finite=False, ensure_2d=False)
+    assert_allclose(X_out, [1, 2, np.nan])
+
+    X_out = check_array(
+        X_int64, force_all_finite=False, ensure_2d=False, dtype=np.float32
```

Comment:
For my education: what does this test compared to the previous test? I was expecting one of the assert statements after this to check that the dtype of the return value is float32 or some other intricacy related to float32s.

Reply:
Thank you for noticing this! I intended to assert the dtype here too. The PR was updated to include them: [`692cfc8` (#25080)](https://github.com/scikit-learn/scikit-learn/pull/25080/commits/692cfc86fdfd6f9ec6b1d8f446dece215af7faf6)